### PR TITLE
Added service worker for "add to home screen" and offline mode.

### PR DESCRIPTION
**Pull request by arodic**
Please test thoroughly before merging :)

I'll add some comments.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -35,6 +39,12 @@
 
     <link rel="icon" href="/images/favicon.ico" />
 
+    <link rel="manifest" href="./manifest.json" />
```
Comment (arodic) [inline index.html:42]: This is the app manifest. I added some basic info and icons but you might wanna add more.

File: index.html [hunk 2]
```diff
@@ -152,8 +162,14 @@
       <script src="./src/simulation.js"></script>
       <script>
 
+        if ('serviceWorker' in navigator) {
+          navigator.serviceWorker.register('./service.js').then(function(serviceWorkerRegistration) {
+            serviceWorkerRegistration.update();
```
Comment (arodic) [inline index.html:167]: Here, I force the update of the service worker on every page load but it updates only if new version has different hash. This makes sure that users don't get stuck with old service worker forever.

File: index.html [hunk 3]
```diff
@@ -152,8 +162,14 @@
...
+          });
+        }
+
         var root = /localhost/.test(window.location)
-          ? './' : '//storage.googleapis.com/cdn.patatap.com/mellow-waves';
+          ? './' : '//storage.googleapis.com/cdn.patatap.com/mellow-waves/';
```
Comment (arodic) [inline index.html:172]: I noticed double slashes in fetch requests so I fixed it.

File: index.html [hunk 4]
```diff
@@ -1059,5 +1075,6 @@
       </script>
 
     </div>
+    <noscript>JavaScript is required to view this page!</noscript>
```
Comment (arodic) [inline index.html:1078]: This is nice to have and ticks one of the accessibility checkboxes.

File: index.html [hunk 5]
```diff
@@ -12,6 +13,9 @@
     <meta name="image" content="http://cornelius.patatap.com/images/thumbnail.jpg">
     <meta name="description" content="Patatap is a portable animation and sound kit. With the touch of a finger create melodies charged with moving shapes. Warning: contains flashing images.">
 
+    <meta name="mobile-web-app-capable" content="yes" />
```
Comment (arodic) [inline index.html:16]: I think these are not necessary if you have the manifest file but I put them just in case...

**Comment by arodic** [comment]
One more thing, when you are debugging app locally, you might want to disable network requests going through service worker. Its in the dev tools application/service worker.

**Comment by jonobr1** [comment]
This is awesome thanks Aki! I'll debug further on phones.

**Pull request merged**